src/testing/storage.zig

```diff
@@ -715,6 +715,13 @@ pub const Storage = struct {
             assert(data_block_header.block_type == .data);
         }
     }
+
+    pub fn transition_to_liveness_mode(storage: *const Storage) void {
+        const storage_options: *Options = @constCast(&storage.options);
```

Comment:
That's not good, that's illegal behavior I think. `const` is not a lint, compiler can rely on stuff not changing in the presence of `const`. Pass `*Storage` here?